chat-widget/src/components/webchatcontainerstateful/webchatcontroller/middlewares/storemiddlewares/attachmentUploadValidatorMiddleware.ts

```diff
@@ -58,8 +60,24 @@ const validateFileExtension = (attachment: any, allowedFileExtensions: string):
     return allExtensions.indexOf(fileExtension) > -1;
 };
 
-const validateFileSize = (attachmentSize: string, maxUploadFileSize: string): boolean => {
-    return (maxUploadFileSize && parseInt(maxUploadFileSize) * MBtoBRatio) > parseInt(attachmentSize);
+const validateFileSize = (attachmentSize: string, maxUploadFileSize: number): boolean => {
+    return (maxUploadFileSize * MBtoBRatio) > parseInt(attachmentSize);
+};
+
+const getmaxUploadFileSize = (maxFileSizeSupportedByDynamicsStr:string, contentType: string): number => {
+    const isFileImage = isImage(contentType);
+    const maxFileSizeSupportedByDynamics = maxFileSizeSupportedByDynamicsStr && parseInt(maxFileSizeSupportedByDynamicsStr) ? parseInt(maxFileSizeSupportedByDynamicsStr) : AMSConstants.maxSupportedFileSize;
+
+    // Takes the smallest max file size configure betteween AMS and Dynamics Config
+    if (isFileImage){
+        return maxFileSizeSupportedByDynamics > AMSConstants.maxSupportedImageSize ? AMSConstants.maxSupportedImageSize : maxFileSizeSupportedByDynamics;
+    } else {
+        return maxFileSizeSupportedByDynamics > AMSConstants.maxSupportedFileSize ? AMSConstants.maxSupportedFileSize: maxFileSizeSupportedByDynamics;
+    }
+};
+
+const isImage = (contentType: string): boolean => {
+    return AMSConstants.supportedImagesMimeTypes.indexOf(contentType) > -1;
```

Comment:
`Array.prototype.includes()` can probably be used instead for readability.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes

Reply:
done